### DISCUSSION TITLE
scroll area height "h-full" make the scroll area gone?

### DISCUSSION THREAD

**Discussion by atticus-lv** [zauberzeug/nicegui#3079] [Q&A] [answered]
### Question

I want the scroll area to auto fit the parent container, but seem it will only fit half height of it...How to make it fill parent? `claasses('h-full')` will make the area gone

video:

https://github.com/zauberzeug/nicegui/assets/64118610/5b04f0ef-145b-40b8-be5a-82b93a5cd0e2

code
```python
from nicegui import ui

with ui.scroll_area().classes('w-full h-[500px]'):
    for i in range(0, 10):
        with ui.card().classes('w-[256px] h-[256px]'):
            ui.label(f'Card {i}')
            ui.button('Edit')
            ui.button('Delete')
            ui.button('Clone')

ui.run(native=True)
```

**Comment by atticus-lv** [comment]
I try `classes('absolute fit inset-0')` will give a smooth interface
![image](https://github.com/zauberzeug/nicegui/assets/64118610/db8b6369-4d53-4bc4-9c1d-585e0f58e935)
However, it must be the most top container. Or it will overwrite others. 
![image](https://github.com/zauberzeug/nicegui/assets/64118610/da458f51-58ab-49f4-a08c-09ca14f0c9b4)
```python
with ui.row().classes('w-full'):
    # search bar
    search = ui.input('Search').props('outlined')
    with search.add_slot('prepend'):
        ui.icon('search')
with ui.scroll_area().classes('absolute fit inset-0'):
    for i in range(0, 5):
        ui.label('test')
```
Any method to fix this?

**Comment by atticus-lv** [comment]
hi @rodja @falkoschindler  ,can I get some help from you?

**Comment by falkoschindler** [comment, marked as answer]
Hi @atticus-lv,

"h-full" let's `ui.scroll_area" get the height from its parent container. But there is no such container. The page itself doesn't automatically fill the viewport, so it's height is 0 by default. If the scroll area uses this height as well, everything collapses to zero height.

You could use something like `.classes('w-full h-[calc(100vh-2rem)]')` to use 100% of the viewport minus 2rem for the padding. If there are other elements taking up some of the vertical space, you might want to force to page to fill the viewport before using "h-full" on the scroll area:
```py
ui.context.client.content.classes('h-screen')

ui.label('Some text')
with ui.scroll_area().classes('w-full h-full'):
    for i in range(0, 10):
        with ui.card().classes('w-[256px] h-[256px]'):
            ui.label(f'Card {i}')
            ui.button('Edit')
            ui.button('Delete')
            ui.button('Clone')
```

**Reply by atticus-lv** [reply]
Thanks, that is a nice solution.

**Reply by atticus-lv** [reply]
But there is still a problem that when this scrollarea use with footer/header. It will show up the scroller of the pager, which make 2 scroll bar in this area. @falkoschindler 
It that a way to hide the page scroller's bar or just make the header over on top of it? In that case I will not define my own scroll area
![image](https://github.com/zauberzeug/nicegui/assets/64118610/9e7fc253-262b-4ee4-82c5-9095dbf7572f)

```python
from nicegui import ui

ui.context.client.content.classes('h-screen')

with ui.header().classes(replace='row items-center pywebview-drag-region', ) as header:
    ui.button(icon='menu').props('flat color=white')

ui.label('Some text')
with ui.scroll_area().classes('w-full h-full'):
    for i in range(0, 10):
        with ui.card().classes('w-[256px] h-[256px]'):
            ui.label(f'Card {i}')
            ui.button('Edit')
            ui.button('Delete')
            ui.button('Clone')
ui.run(native=True)
```

**Reply by falkoschindler** [reply]
You somehow need to account for the header when setting the content height. One way is to simply subtract its height:
```py
ui.context.client.content.classes('h-[calc(100vh-2rem)]')
```

But I found another alternative [in this comment](https://github.com/zauberzeug/nicegui/discussions/1314) avoiding such magic numbers:
```py
ui.query('.nicegui-content').classes('w-full')
ui.query('.q-page').classes('flex')
```

**Reply by vmlstandard** [reply]
What if I do not want to have 2 vertical scrollbars, just one:

```py
from nicegui import ui

ui.context.client.content.classes('h-screen')

with ui.header().classes(replace='row items-center pywebview-drag-region', ) as header:
    ui.button(icon='menu').props('flat color=white')

ui.label('Some text')
for i in range(0, 10):
    with ui.card().classes('w-[256px] h-[256px]'):
        ui.label(f'Card {i}')
        ui.button('Edit')
        ui.button('Delete')
        ui.button('Clone')

# ui.footer()
ui.run()
```


if I throw a footer into the mix, the scroll gets messed up: uncomment the # ui.footer() and you will see

**Reply by balex89** [reply]
> But I found another alternative [in this comment](https://github.com/zauberzeug/nicegui/discussions/1314) avoiding such magic numbers:

Talking about _magic numbers_, looks like Quazar itself dynamically updates `padding-top` style prop for `q-page-container`  to exactly fit current `header` height in pixels so that `q-page` and `header` does not overlap: https://github.com/quasarframework/quasar/blob/dev/ui/src/components/page/QPageContainer.js#L25

So with similar problem i ended up computing `header` height on window `load` and `resize` (via JS of course). Then saving it to a variable like `--header-size` and using it in `calc()` functions for styles, e.g. for a `scroll_area` to fit remaining space. 
Not sure if this is optimal, but works for me!

**Reply by vmlstandard** [reply]
Pretty clever, thank you for the idea!
